generate_corepop_compat_report.py

```diff
@@ -0,0 +1,110 @@
+"""
+generate a markdown report of which corepops are compatible with which platforms
+"""
+import subprocess
+import functools
+import docker
+from typing import List, Tuple
+from pathlib import Path
+from concurrent import futures
+
+
+DOCKER_CLIENT = docker.from_env()
+OS_NAME = subprocess.check_output(["uname", "-s"]).decode('utf-8').lower().strip()
+ARCH = subprocess.check_output(["uname", "-m"]).decode('utf-8').strip()
+
+DEPS = {
+    'ubuntu': ['libc6', 'libncurses5', 'libncurses5-dev', 'libstdc++6', 'libxext6', 'libx11-6', 'libx11-dev', 'libxt-dev', 'libmotif-dev'],
+    'fedora': ['ncurses-devel', 'openmotif-devel', '']
+}
+
+CONTAINERS = [
+    ('ubuntu', '21.04'),
+    ('ubuntu', '20.04'),
+    ('ubuntu', '18.04'),
+    ('ubuntu', '16.04'),
+    ('debian', 'unstable-slim'),
+    ('debian', 'testing-slim'),
+    ('debian', 'stable-slim'),
+    ('debian', 'oldstable-slim'),
+    ('debian', 'oldoldstable'),
+    ('fedora', '34'),
+    ('fedora', '33'),
+    ('fedora', '32'),
+    ('centos', '8'),
+    ('centos', '7'),
+]
+
+
+def unpack_tuple_args(fn):
+    @functools.wraps(fn)
+    def wrapped(args):
+        return fn(*args)
+    return wrapped
+
+
+def list_available_corepops() -> List[Path]:
+    return sorted([p for p in (Path(OS_NAME) / ARCH).iterdir()
+                   if p.name.endswith('.corepop')])
```

Comment:
Non-semantic indentation at random level. Depending on your preferences for compactness
```
return sorted([
    p for p in (Path(OS_NAME) / ARCH).iterdir()
        if p.name.endswith('.corepop')
])
```
or
```
return sorted(
    [
        p 
        for p in (Path(OS_NAME) / ARCH).iterdir()
            if p.name.endswith('.corepop')
    ]
    )
```